specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2024-09-01/dataConnectorDefinitions.json

```diff
@@ -0,0 +1,706 @@
+          },
+          {
+            "in": "body",
+            "name": "connectorDefinitionInput",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DataConnectorDefinition"
+            },
+            "description": "The data connector definition",
+            "x-ms-parameter-location": "method"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Updated",
+            "schema": {
+              "$ref": "#/definitions/DataConnectorDefinition"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/DataConnectorDefinition"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "delete": {
+        "x-ms-examples": {
+          "Delete data connector definition": {
+            "$ref": "./examples/dataConnectorDefinitions/DeleteDataConnectorDefinitionById.json"
+          }
+        },
+        "tags": [
+          "ConnectorDefinitions"
+        ],
+        "description": "Delete the data connector definition.",
+        "operationId": "DataConnectorDefinitions_Delete",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/dataConnectorDefinitionName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success"
+          },
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ConnectorDefinitionsAvailability": {
+      "description": "The exposure status of the connector to the customers.",
+      "type": "object",
+      "properties": {
+        "status": {
+          "$ref": "#/definitions/AvailabilityStatus"
+        },
+        "isPreview": {
```

Comment:
>isPreview [](http://example.com/codeflow?start=9&length=9)

ARM recommends enum instead of boolean. Consider something like stage: { Preview, Stable }. You can add future values later, e.g. Private, Deprecated, Removed, etc.

Reply:
@mentat9  
The reason we didnt use Enum, was to save backword compatability for the customers, since thery are using today the same object in another API. This was the Product Manager desicion. 

This API was review in preview version. 
we want to have exactly the same api  in Stable version.